src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -177,6 +177,9 @@ public virtual void WriteResponseHeaders([NotNull] OutputFormatterContext contex
                 throw new InvalidOperationException(Resources.FormatOutputFormatterNoMediaType(GetType().FullName));
             }
 
+            // Clone the media type as we don't want it to affect the next request
+            selectedMediaType = new MediaTypeHeaderValue(selectedMediaType.ToString());
```

Comment:
I am a bit unclear about what you are saying here...I thought this would be the safest as we clone as early as possible so that any code later can only see the cloned one...